extendeddatacrossword.go

```diff
@@ -178,7 +178,13 @@ func (eds *ExtendedDataSquare) solveCrosswordRow(
 
 	// Insert rebuilt shares into square.
 	for c, s := range rebuiltShares {
-		eds.SetCell(uint(r), uint(c), s)
```

Comment:
previously this was silently ignoring an error! Thanks to errcheck, we are forced to check the error return type and propagate it which resulted in:

```
--- FAIL: TestRepairExtendedDataSquare (0.00s)
    --- FAIL: TestRepairExtendedDataSquare/leopard (0.00s)
        --- FAIL: TestRepairExtendedDataSquare/leopard/MaximumErasures (0.00s)
            /Users/rootulp/git/rootulp/celestia/rsmt2d/extendeddatacrossword_test.go:60: unexpected err while repairing data square: cannot set cell (3, 2) as it already has a value 05050505050505050505050505050505050505050505050505050505050505050505050505050505050505050505050505050505050505050505050505050505, codec: :leopard
```

To resolve, 2b4078bd7d01f8cbf125be4c533d09cb33c55072 checks if the cellToSet is nil prior to the attempt to `SetCell`